### PR TITLE
Api extensions

### PR DESCRIPTION
**Pull request by Zengo-one**

### CODE REVIEW

File: api/RestaurantBusiness.BLL/Services/RestaurantService.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+        public async Task<IEnumerable<RestaurantDto>> GetRestaurants()
+        {
+            var restaurants = await _restaurantRepository.GetAllItemsAsync(null);
+            var restaurantDtos = new List<RestaurantDto>();
+            foreach(var restaurant in restaurants)
+            {
+                var restaurantDto = _mapper.Map<RestaurantDto>(restaurant);
+                restaurantDto.Menu = new List<Food>();
+                restaurantDto.Menu.AddRange(await _foodRepository.GetAllItemsAsync(f => f.RestaurantId == restaurant.Id));
```
Comment (skynetigor) [inline api/RestaurantBusiness.BLL/Services/RestaurantService.cs:60]: Can you make retrieving of menus and addresses in parallel using LINQ?

File: api/RestaurantBusiness.BLL/Services/RestaurantService.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                try
+                {
+                    restaurantDto.Address = await _addressRepository.GetItemAsync(restaurant.AddressId);
+
+                }
+                catch(Exception ex)
+                {
+
```
Comment (skynetigor) [inline api/RestaurantBusiness.BLL/Services/RestaurantService.cs]: Is this catch needed?

File: api/RestaurantBusiness.BLL/Services/RestaurantService.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            var restaurantDto = _mapper.Map<RestaurantDto>(restaurant);
+
+            return restaurantDto;
+        }
+
+        public async Task<IEnumerable<RestaurantDto>> GetRestaurants()
+        {
+            var restaurants = await _restaurantRepository.GetAllItemsAsync(null);
+            var restaurantDtos = new List<RestaurantDto>();
+            foreach(var restaurant in restaurants)
```
Comment (skynetigor) [inline api/RestaurantBusiness.BLL/Services/RestaurantService.cs:56]: Add spaces before/after foreach, if, try/catch statements.

File: api/RestaurantBusiness.BLL/Services/RestaurantService.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        {
+            var restaurant = await _restaurantRepository.GetItemAsync(id);
+            var restaurantDto = _mapper.Map<RestaurantDto>(restaurant);
+
+            return restaurantDto;
+        }
+
+        public async Task<IEnumerable<RestaurantDto>> GetRestaurants()
+        {
+            var restaurants = await _restaurantRepository.GetAllItemsAsync(null);
```
Comment (skynetigor) [inline api/RestaurantBusiness.BLL/Services/RestaurantService.cs:54]: Why not use optional param in _restaurantRepository.GetAllItemsAsync() ?

File: api/RestaurantBusiness.BLL/Services/RestaurantService.cs [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        public async Task CreateRestaurant(RestaurantDto restaurantDto)
+        {
+            var restaurant = _mapper.Map<Restaurant>(restaurantDto);
+            restaurant.Id = Guid.NewGuid().ToString();
+            await _restaurantRepository.CreateItemAsync(restaurant);
+
+            foreach(var food in restaurantDto.Menu)
+            {
+                food.RestaurantId = restaurant.Id;
+                await _foodRepository.CreateItemAsync(food);
```
Comment (skynetigor) [inline api/RestaurantBusiness.BLL/Services/RestaurantService.cs:40]: You can run it in parallel Tasks.

File: api/RestaurantBusiness.DAL/Repositories/AddressRepository.cs [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        {
+            item.Id = Guid.NewGuid().ToString();
+            await _addressContainer.CreateItemAsync(item);
+        }
+
+        public async Task CreateSeveralItems(IEnumerable<Address> items)
+        {
+            foreach(var item in items)
+            {
+                await _addressContainer.CreateItemAsync(item, new PartitionKey(item.Country));
```
Comment (skynetigor) [inline api/RestaurantBusiness.DAL/Repositories/AddressRepository.cs:34]: Run in parallel

File: api/RestaurantBusiness.DAL/Repositories/AddressRepository.cs [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+            var addresses = new List<Address>();
+            addresses.AddRange(await addressQueryable.ToFeedIterator().ReadNextAsync());
+
+            return addresses;
+        }
+
+        public async Task<Address> GetItemAsync(string id)
+        {
+            return await _addressContainer.ReadItemAsync<Address>(id, new PartitionKey("Ukraine"));
```
Comment (skynetigor) [inline api/RestaurantBusiness.DAL/Repositories/AddressRepository.cs]: Why is it hardcoded?

File: api/RestaurantBusiness.DAL/Repositories/FoodRepository.cs [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        {
+            item.Id = Guid.NewGuid().ToString();
+            await _foodContainer.CreateItemAsync(item);
+        }
+
+        public async Task CreateSeveralItems(IEnumerable<Food> items)
+        {
+            foreach(var item in items)
+            {
+                await _foodContainer.CreateItemAsync(item);
```
Comment (skynetigor) [inline api/RestaurantBusiness.DAL/Repositories/FoodRepository.cs:34]: Please use parallel tasks whenever it's possible

File: api/RestaurantBusiness.BLL/Services/RestaurantService.cs (hunk 2 above)
Comment (Zengo-one) [reply api/RestaurantBusiness.BLL/Services/RestaurantService.cs]: It's outdated

**Pull request merged**